sleap/gui/commands.py

```diff
@@ -3470,13 +3481,230 @@ def do_action(context: CommandContext, params: dict):
         if rls is not None:
             context.openWebsite(rls.url)
 
+class ExportClipVideo(AppCommand):
+    @staticmethod
+    def do_action(context: CommandContext, params: dict):
+        """
+        Executes the export action for a video clip and its labels.
 
-def copy_to_clipboard(text: str):
-    """Copy a string to the system clipboard.
+        Args:
+            context (CommandContext): The application context containing the current state.
+            params (dict): Parameters for the export, including:
+                - 'filename' (str): The path to save the exported video.
+                - 'fps' (int): Frames per second for the exported video.
+                - 'open_when_done' (bool): Whether to open the video file after exporting.
+
+        Raises:
+            ValueError: If the frame range is invalid or no clip is selected.
+            RuntimeError: If there are issues with video writing or saving labels.
+        """
+        # Extract video and labels from context
+        video = context.state["video"]
+        labels = context.state["labels"]
 
-    Args:
-        text: String to copy to clipboard.
-    """
-    clipboard = QtWidgets.QApplication.clipboard()
-    clipboard.clear(mode=clipboard.Clipboard)
-    clipboard.setText(text, mode=clipboard.Clipboard)
+        # Ensure frame range is set; default to all frames if None
+        frame_range = context.state.get("frame_range", (0, video.frames))
```

Comment:
Maybe we say 
```suggestion
        frame_range = context.state.get("frame_range", None)
```
and raise the no valid clip range error below if `frame_range is None`?